testament/testament.nim

```diff
@@ -821,14 +1048,21 @@ proc codegenCheck(
       given.msg = "generated code size: " & $contents.len
       expectedMsg = "max allowed size: " & $spec.maxCodeSize
   except ValueError:
+    # NOTE REFACTOR that's an excellent candidate for separation into the
+    # test report kind.
     given.err = reInvalidPeg
     msg Undefined: getCurrentExceptionMsg()
   except IOError:
     given.err = reCodeNotFound
     msg Undefined: getCurrentExceptionMsg()
 
-proc compilerOutputTests(run: var TestRun, given: var TSpec; r: var TResults) =
-  ## Test output of the compiler for correctness
+proc compilerOutputTests(run: TestRun, given: var TSpec; r: var TResults) =
+  ## Test output of the compiler for correctness and add comparison results
+  ## to the `TResults` output.
+  # REFACTOR into a functio that returns result of the output test
```

Comment:
This could go either way, because some tests do a compile and run vs just a compile. Accumulating results while also supporting the latter in the same structure vs separate structures and stitching them back together.